workflow/util/pod_name.go

```diff
@@ -79,8 +79,10 @@ func ensurePodNamePrefixLength(prefix string) string {
 // given workflow
 func GetWorkflowPodNameVersion(wf *v1alpha1.Workflow) PodNameVersion {
 	annotations := wf.GetAnnotations()
-	version := annotations[common.AnnotationKeyPodNameVersion]
-
+	version, ok := annotations[common.AnnotationKeyPodNameVersion]
```

Comment:
Is the `default` in the switch case below still valid?  Otherwise this could potentially be rewritten so that `GetPodNameVersion()` is the `default` case?

Reply:
I was tempted to implement this fix that way but that default switch case will be hit if the annotation is invalid. That is a different behaviour to the GetPodNameVersion call. 

But yeah I am on the fence on this about what the "right behaviour" is.  This is why I went with the current approach, at least it will always be consistent when the user makes a wrong annotation (default to v2).

Reply:
Ok, sounds good!